### PR TITLE
MTDSA-6745 Added Response Hateoas Links+Body for "UpdateStateBenefits"

### PR DESCRIPTION
**Pull request by ngangaNjiraini**

### CODE REVIEW

File: app/v1/hateoas/HateoasLinks.scala [hunk 1]
```diff
@@ -66,19 +69,25 @@ trait HateoasLinks {
...
   // State benefits Hateoas
+  def addStateBenefits(appConfig: AppConfig, nino: String, taxYear: String): Link =
+    Link(
+      href = baseUri(appConfig, nino, taxYear),
+      method = POST,
+      rel = ADD_STATE_BENEFIT
+    )
 
-  def updateStateBenefits(appConfig: AppConfig, nino: String, taxYear: String, benefitId: String): Link =
+  def updateStateBenefit(appConfig: AppConfig, nino: String, taxYear: String, benefitId: String): Link =
```
Comment (narendravijayarao) [inline app/v1/hateoas/HateoasLinks.scala:79]: Why did you change this??

File: app/v1/models/response/AddStateBenefitsResponse.scala [hunk 2]
```diff
@@ -32,8 +32,8 @@ object AddStateBenefitsResponse extends HateoasLinks {
       import data._
       Seq(
         listStateBenefits(appConfig, nino, taxYear),
-        updateStateBenefits(appConfig, nino, taxYear, benefitId),
-        deleteStateBenefits(appConfig, nino, taxYear, benefitId)
+        updateStateBenefit(appConfig, nino, taxYear, benefitId),
```
Comment (narendravijayarao) [inline app/v1/models/response/AddStateBenefitsResponse.scala:35]: the endpoint is **Update state benefits**

Comment (MarkAKelly) [reply app/v1/models/response/AddStateBenefitsResponse.scala:35]: There are two endpoints, Update State Benefit, and Update State Benefit Amounts

File: app/v1/hateoas/HateoasLinks.scala (hunk 1 above)
Comment (BetaDraconis) [reply app/v1/hateoas/HateoasLinks.scala:79]: the endpoint is called `Update State Benefit` not `Update State Benefit**s**`
